libs/ui-toolkit/src/components/tabs/tabs.tsx

```diff
@@ -17,24 +18,40 @@ export const Tabs = ({
   onValueChange,
   ...props
 }: TabsProps) => {
+  // Bunch of refs in order to detect wrapping in side the tabs so that we
+  // can apply a bg color
+  const wrapperRef = useRef<HTMLDivElement | null>(null);
```

Comment:
Adding the refs and the resize observer is only needed to set the background color when the menu wraps underneath the tabs. This very much annoys me. LMK if you have a better idea

Reply:
the menu seems to be “wrapped” only if it’s not the direct sibling of [role=tablist], hence in this case:
```
className={classNames(
            'flex-1 p-1',
            'bg-vega-clight-700 dark:bg-vega-cdark-700',
            'peer-[[role="tablist"]]:bg-inherit dark:peer-[[role="tablist"]]:bg-inherit'
          )
```

would do the same as what you're trying to do with refs

Reply:
```
diff --git a/libs/ui-toolkit/src/components/tabs/tabs.tsx b/libs/ui-toolkit/src/components/tabs/tabs.tsx
index d20720556..960f07f08 100644
--- a/libs/ui-toolkit/src/components/tabs/tabs.tsx
+++ b/libs/ui-toolkit/src/components/tabs/tabs.tsx
@@ -2,11 +2,10 @@ import * as TabsPrimitive from '@radix-ui/react-tabs';
 import {
   useLocalStorageSnapshot,
   getValidItem,
-  useResizeObserver,
 } from '@vegaprotocol/react-helpers';
 import classNames from 'classnames';
 import type { ReactElement, ReactNode } from 'react';
-import { Children, isValidElement, useRef, useState } from 'react';
+import { Children, isValidElement, useState } from 'react';
 export interface TabsProps extends TabsPrimitive.TabsProps {
   children: (ReactElement<TabProps> | null)[];
 }
@@ -25,19 +24,6 @@ export const Tabs = ({
     return children.find((v) => v)?.props.id;
   });
 
-  // Bunch of refs in order to detect wrapping in side the tabs so that we
-  // can apply a bg color
-  const wrapperRef = useRef<HTMLDivElement | null>(null);
-  const tabsRef = useRef<HTMLDivElement | null>(null);
-  const menuRef = useRef<HTMLDivElement | null>(null);
-  const [wrapped, setWrapped] = useState(() =>
-    isWrapped(tabsRef.current, menuRef.current)
-  );
-
-  useResizeObserver(wrapperRef.current, () => {
-    setWrapped(isWrapped(tabsRef.current, menuRef.current));
-  });
-
   return (
     <TabsPrimitive.Root
       {...props}
@@ -45,14 +31,10 @@ export const Tabs = ({
       onValueChange={onValueChange || setActiveTab}
       className="h-full grid grid-rows-[min-content_1fr]"
     >
-      <div
-        ref={wrapperRef}
-        className="flex flex-wrap justify-between border-b border-default min-w-0"
-      >
+      <div className="flex flex-wrap justify-between border-b border-default min-w-0">
         <TabsPrimitive.List
-          className="flex flex-nowrap overflow-visible"
+          className="flex flex-nowrap overflow-visible peer"
           role="tablist"
-          ref={tabsRef}
         >
           {Children.map(children, (child) => {
             if (!isValidElement(child) || child.props.hidden) return null;
@@ -86,20 +68,18 @@ export const Tabs = ({
           })}
         </TabsPrimitive.List>
         <div
-          ref={menuRef}
-          className={classNames('flex-1 p-1', {
-            'bg-vega-clight-700 dark:bg-vega-cdark-700': wrapped,
-            '': wrapped,
-          })}
+          className={classNames(
+            'flex-1 p-1',
+            'bg-vega-clight-700 dark:bg-vega-cdark-700',
+            'peer-[[role="tablist"]]:bg-inherit dark:peer-[[role="tablist"]]:bg-inherit'
+          )}
         >
           {Children.map(children, (child) => {
             if (!isValidElement(child) || child.props.hidden) return null;
             return (
               <TabsPrimitive.Content
                 value={child.props.id}
-                className={classNames('flex flex-nowrap gap-1', {
-                  'justify-end': !wrapped,
-                })}
+                className={classNames('flex flex-nowrap gap-1', 'justify-end')}
               >
                 {child.props.menu}
               </TabsPrimitive.Content>
@@ -162,13 +142,3 @@ export const LocalStoragePersistTabs = ({
     />
   );
 };
-
-const isWrapped = (
-  tabs: HTMLDivElement | null,
-  menu: HTMLDivElement | null
-) => {
-  if (!tabs || !menu) return;
-  const listRect = tabs.getBoundingClientRect();
-  const menuRect = menu.getBoundingClientRect();
-  return menuRect.y > listRect.y;
-};

```

Reply:
Ignore me, I've been resizing the whole window instead of a pane

Reply:
This won't work when you resize the pane.. window resize is handled by the switch from trade-grid to trade-panels